client/packages/system/src/Encounter/api/hooks/document/useEncounter.ts

```diff
@@ -7,15 +7,6 @@ export const useEncounter = () => {
   const id = useEncounterId();
 
   return {
-    ...useQuery(
-      api.keys.detail(id),
-      () => api.get.byId(id),
-      // Don't refetch when the edit modal opens, for example. But, don't cache data when this query
-      // is inactive. For example, when navigating away from the page and back again, refetch.
-      {
-        refetchOnMount: false,
-        cacheTime: 0,
```

Comment:
Would it be an option to check for undefined in the `byId` api call and throw an error there?

Reply:
that's where the error is thrown - unless you mean somewhere else?
```
        throw new Error('Could not find encounter');
```

After re-looking, the react query does end up in an error state because of this.. but it does appear to differ from the case where the error is throw in the network layer of the call. I still believe that the query implementation should work, so will revisit this. it bugs me that we have to implement state in the detail view to handle this